proposals/src/target_system.rs

```diff
@@ -111,6 +111,15 @@ impl TargetSystem {
         }
     }
 
+    /// Get smart contract TargetSystem details
+    #[must_use]
+    pub fn into_contract_address(self) -> [u8; 20] {
+        match self {
+            TargetSystem::ContractAddress(address) => address,
+            _ => [0u8; 20],
+        }
+    }
```

Comment:
I do not prefer having a function here without defined behaviour, it does not at least explain (for the caller from the function signature) on what this function would do in the case if the `self` is not a `TargetSystem::ContractAddress`?  maybe something like this could work better: 
```suggestion
    /// Get smart contract address from the target system.
    /// Returns [`None`] if the target system is not a contract address.
    #[must_use]
    pub fn into_contract_address(self) -> Option<[u8; 20]> {
        match self {
            TargetSystem::ContractAddress(address) => Some(address),
            _ => None,
        }
    }
```

Or maybe:

```suggestion
    /// Get smart contract address from the target system.
    /// Returns `[0; 20]` if the target system is not a contract address.
    #[must_use]
    pub fn into_contract_address_or_default(self) -> [u8; 20] {
        match self {
            TargetSystem::ContractAddress(address) => address,
            _ => [0; 20],
        }
    }
```

So you see here, the function signature and the comment/docs says it all to the caller that if the target system is not a contract address.

Reply:
Updated